crates/transcode/src/lib.rs

```diff
@@ -331,6 +361,21 @@ impl ContractMessageTranscoder {
             args.push((Value::String(name), value));
         }
 
+        if !data.is_empty() {
+            let arg_list_string: String = args
+                .iter()
+                .fold(format!("`{}`", msg_spec.label()), |init, arg| {
+                    format!("{}, `{}`", init, arg.0)
+                });
+            let encoded_bytes = util::encode_bytes(data);
+            return Err(anyhow::anyhow!(
+                "input length was longer than expected by {} byte(s).\nManaged to decode {} but `{}` bytes were left unread",
+                data.len(),
+                arg_list_string,
+                encoded_bytes
+            ))
+        }
```

Comment:
The code to produce this error message is duplicated 3 times so would be good to either extract to a function, or perform one time at a higher level (e.g. in `decode.rs`).

Reply:
It would require me to push the error to the higher level. I'll see what works the best